secure_message/resources/messages.py

```diff
@@ -39,6 +39,11 @@ def post(self):
         message = self._validate_post_data(post_data)
 
         if message.errors == {}:
+            if message.thread_id:
+                conversation_metadata = Retriever.retrieve_thread_metadata(message.thread_id)
```

Comment:
I think we need to be clear in our use of conversation vs thread. I think anything related to the new table should refer to conversation rather than thread - so this would be `retrieve_conversation_metadata`. This also applies to all the other places this occurs in this pull request.